File: packages/lang-c/tests/fixtures/simple/greeter.c
/**
 * Tiny greeter — used as the lang-c "simple" fixture.
 */
#include <stdio.h>
#include "greeter.h"

/**
 * Print "Hello, <name>!" to stdout.
 */
void greet(const char* name) {
  printf("Hello, %s!\n", name);
}

int main(int argc, char** argv) {
  const char* who = argc > 1 ? argv[1] : "world";
  greet(who);
  return 0;
}

File: packages/lang-java/tests/fixtures/simple/Greeter.java
package com.example.simple;

import java.util.Objects;

/**
 * A tiny greeter used as the simple-tier fixture for {@code lang-java}.
 *
 * Exercises: top-level class, fields (final & non-final), constructor,
 * method calls, a static method, and a leading Javadoc on both the
 * class and its members.
 */
public final class Greeter {
    public static final String DEFAULT_PREFIX = "Hello";

    private final String prefix;

    /** Default greeter using {@link #DEFAULT_PREFIX}. */
    public Greeter() {
        this(DEFAULT_PREFIX);
    }

    /** Greeter with a custom prefix. */
    public Greeter(String prefix) {
        this.prefix = Objects.requireNonNull(prefix, "prefix");
    }

    /**
     * Greet a name.
     *
     * @param name the name to greet
     * @return the greeting message
     */
    public String greet(String name) {
        Objects.requireNonNull(name, "name");
        return prefix + ", " + name + "!";
    }

    public static Greeter english() {
        return new Greeter("Hi");
    }
}
